node/narwhal/src/worker.rs

```diff
@@ -477,10 +479,228 @@ mod prop_tests {
+        // Initialize the storage.
+        let storage = Storage::<CurrentNetwork>::new(committee.clone(), 1);
+        let gateway = MockGateway::default();
+        let mut mock_ledger = MockLedgerService::default();
+        mock_ledger.expect_contains_transmission().returning(|_| Ok(false));
+        mock_ledger.expect_check_solution_basic().returning(|_, _| Ok(()));
+        let ledger: Ledger<CurrentNetwork> = Arc::new(mock_ledger);
+        let worker = Worker::new(1, Arc::new(gateway), storage, ledger, Default::default()).unwrap();
+        let data = |rng: &mut TestRng| Data::Buffer(Bytes::from((0..512).map(|_| rng.gen::<u8>()).collect::<Vec<_>>()));
+        let transmission_id = TransmissionID::Solution(PuzzleCommitment::from_g1_affine(rng.gen()));
+        let peer_ip = SocketAddr::from(([127, 0, 0, 1], 1234));
+        let transmission = Transmission::Solution(data(rng));
+        worker.process_transmission_from_peer(peer_ip, transmission_id, transmission.clone());
+        assert!(worker.contains_transmission(transmission_id));
+        assert!(worker.ready.contains(transmission_id));
+        assert_eq!(worker.get_transmission(transmission_id), Some(transmission));
+        // Take the transmission from the ready set.
+        let transmission: Vec<_> = worker.take_candidates(1).await.collect();
+        assert_eq!(transmission.len(), 1);
+        assert!(!worker.ready.contains(transmission_id));
+    }
+
+    #[tokio::test]
+    async fn test_send_transmission() {
+        let rng = &mut TestRng::default();
+        // Sample a committee.
+        let committee = snarkos_node_narwhal_committee::test_helpers::sample_committee(rng);
+        // Initialize the storage.
+        let storage = Storage::<CurrentNetwork>::new(committee.clone(), 1);
+        let mut gateway = MockGateway::default();
+        gateway.expect_send().returning(|_, _| ());
+
+        let ledger: Ledger<CurrentNetwork> = Arc::new(MockLedgerService::new());
+        let worker = Worker::new(1, Arc::new(gateway), storage, ledger, Default::default()).unwrap();
+
+        let transmission_id = TransmissionID::Solution(PuzzleCommitment::from_g1_affine(rng.gen()));
+        let worker_ = worker.clone();
+        let peer_ip = SocketAddr::from(([127, 0, 0, 1], 1234));
+        let _ = worker_.send_transmission_request(peer_ip, transmission_id).await;
+        assert!(worker.pending.contains(transmission_id));
+        let peer_ip = SocketAddr::from(([127, 0, 0, 1], 1234));
+        // Fake the transmission response.
+        worker.finish_transmission_request(peer_ip, TransmissionResponse {
+            transmission_id,
+            transmission: Transmission::Solution(Data::Buffer(Bytes::from(vec![0; 512]))),
+        });
+        // Check the transmission was removed from the pending set.
+        assert!(!worker.pending.contains(transmission_id));
+    }
+
+    #[tokio::test]
+    async fn test_process_solution_ok() {
```

Comment:
applies to this one and the rest of the tests - they need some newlines that at least separate the setup part from the actual test bits